webrender/res/prim_shared.glsl

```diff
@@ -661,9 +662,10 @@ VertexInfo write_transform_vertex(RectWithSize local_segment_rect,
     vec2 device_pos = world_pos.xy / world_pos.w * uDevicePixelRatio;
     vec2 task_offset = task.common_data.task_rect.p0 - task.content_origin;
 
-#ifdef FORCE_NO_PERSPECTIVE
-    world_pos.w = 1.0;
-#endif
+    // Force w = 1, if we don't want perspective interpolation (for
+    // example, drawing a screen-space quad on an element with a
+    // perspective transform).
+    world_pos.w = mix(1.0, world_pos.w, do_perspective_interpolation);
```

Comment:
wouldn't it be more straightforward to just do `if !do_perspective_interpolation {..}`?

Reply:
Force of habit to avoid things that look like ```if``` statements even though I'm sure all shader compilers would probably find the ```if``` easier to convert to an optimal implementation. I'll change that as a follow up since this is already r+'ed.